core-primitives/node-api/metadata/src/lib.rs

```diff
@@ -40,6 +39,14 @@ pub mod metadata_mocks;
 pub trait NodeMetadataTrait: TeerexCallIndexes + SidechainCallIndexes {}
 impl<T: TeerexCallIndexes + SidechainCallIndexes> NodeMetadataTrait for T {}
 
+impl TryFrom<NodeMetadata> for Metadata {
+	type Error = ();
```

Comment:
you can use the `crate::error::Error` here and then return the `Error::MetadataNotSet`